### PR TITLE
Fix operation name in SearchController and correct nullability for SearchHint

### PR DESCRIPTION
**Pull request by nielsvanvelzen**
_Untested, I don't have a proper test environment_

**Changes**
- Fix operation name in SearchController
  Operation names should be unique across the application. The search thingy used `get` as name which is not that unique of a name
- Correct nullability for SearchHint
  - Basically looked at the nullability of BaseItemDto (which all info is copied from) and what's conditionally set in the controller to determine nullability
- Use BaseItemKind enum in SearchHint
  - GetClientTypeName returns basically the same but as a string

**Issues**

### CODE REVIEW

File: MediaBrowser.Model/Search/SearchHint.cs [hunk 1]
```diff
@@ -1,8 +1,8 @@
-#nullable disable
-#pragma warning disable CS1591
+#pragma warning disable CS0618
```
Comment (Bond-009) [inline MediaBrowser.Model/Search/SearchHint.cs:1]: Why is this needed? `ItemId` is referenced here, is it?

**Review by Bond-009 — APPROVED** [review]

File: MediaBrowser.Model/Search/SearchHint.cs (hunk 1 above)
Comment (nielsvanvelzen) [reply MediaBrowser.Model/Search/SearchHint.cs:1]: Id and ItemId are the same, from SearchController:

```
            // legacy
            result.ItemId = result.Id;
```

So I want to mark ItemId as obsolete so it generates the `deprecated: true` field in the OpenAPI specification. Can't figure out how to disable the error during build though..

**Review by crobibero — APPROVED** [review]

**Comment by Bond-009** [comment]
Build seems to fail?

File: MediaBrowser.Model/Search/SearchHint.cs (hunk 1 above)
Comment (Bond-009) [reply MediaBrowser.Model/Search/SearchHint.cs:1]: I would expect the warning to be inside the `SearchController` file :thinking:

**Comment by nielsvanvelzen** [comment]
Yeah build was failing because of the [Obsolete] thingy, fixed now.

File: MediaBrowser.Model/Search/SearchHint.cs (hunk 1 above)
Comment (nielsvanvelzen) [reply MediaBrowser.Model/Search/SearchHint.cs]: Yeah that's why it didn't work

**Review by Bond-009 — APPROVED** [review]
